testpar/t_mdset.c

```diff
@@ -1456,21 +1456,17 @@ read_attribute(hid_t obj_id, int this_type, int num)
     if (this_type == is_group) {
         HDsprintf(attr_name, "Group Attribute %d", num);
         aid = H5Aopen(obj_id, attr_name, H5P_DEFAULT);
-        if (MAINPROCESS) {
```

Comment:
This routine is only used in a single test (multiple_group_read) in t_mdset.c. That test sets up a FAPL that has collective metadata reads enabled, but this does the attribute read on rank 0 only.

Reply:
could be a cache side-effect for native?

Reply:
Maybe. I'm not sure why this hasn't seemed to be a problem in HDF5.